AppCenter/AppCenterTests/MSOneCollectorChannelDelegateTests.m

```diff
@@ -251,6 +251,38 @@ - (void)testDidEnqueueLogToOneCollectorChannelWhenLogHasTargetTokensAndLogIsNotC
                                }];
 }
 
+- (void)testDidEnqueueLogToOneCollectorChannelSynchronously {
+
+  // If
+  id<MSChannelUnitProtocol> channelUnitMock = OCMProtocolMock(@protocol(MSChannelUnitProtocol));
+  OCMStub([channelUnitMock configuration]).andReturn(self.baseUnitConfig);
+  id<MSChannelGroupProtocol> channelGroupMock = OCMProtocolMock(@protocol(MSChannelGroupProtocol));
+  id<MSChannelUnitProtocol> oneCollectorChannelUnitMock = OCMProtocolMock(@protocol(MSChannelUnitProtocol));
+  OCMStub(oneCollectorChannelUnitMock.logsDispatchQueue).andReturn(self.logsDispatchQueue);
+  OCMStub([channelGroupMock addChannelUnitWithConfiguration:OCMOCK_ANY withIngestion:OCMOCK_ANY]).andReturn(oneCollectorChannelUnitMock);
+  NSMutableSet *transmissionTargetTokens = [NSMutableSet new];
+  [transmissionTargetTokens addObject:@"fake-transmission-target-token"];
+  MSCommonSchemaLog *commonSchemaLog = [MSCommonSchemaLog new];
+  id<MSMockLogWithConversion> mockLog = OCMProtocolMock(@protocol(MSMockLogWithConversion));
+  OCMStub([mockLog toCommonSchemaLogs]).andReturn(@[ commonSchemaLog ]);
+  OCMStub(mockLog.transmissionTargetTokens).andReturn(transmissionTargetTokens);
+
+  /*
+   * Make sure that the common schema log is enqueued synchronously by putting a task on the log queue that won't return
+   * by the time verify is called.
+   */
+  dispatch_async(oneCollectorChannelUnitMock.logsDispatchQueue, ^{
+    sleep(1);
```

Comment:
Why sleep though?

Reply:
We chose sleep to simulate the behavior when doing something like configuring an authentication provider.  We could do that in this block to mirror the case for which the unintended behavior was observed, but it isn't necessarily limited to that scenario.

It is kind of a weird choice; if you have a suggestion to make it clearer it would be appreciated 😄

Reply:
We could use a semaphore, or equivalent, to guarantee order of operations during the test instead of sleeping.